strawberryfields/utils.py

```diff
@@ -673,3 +713,415 @@ def f_proxy(*args):
             return self
 
         return f_proxy
+
+
+#=================================================
+# Engine functions
+#=================================================
+
+
+def is_unitary(engine):
+    """Returns True if every command in the queue is of type :class:`strawberryfields.ops.Gate`.
+
+    Args:
+        engine (Engine): a Strawberry Fields engine instance
+
+    Returns:
+        bool: returns True if all queued operations are unitary
+    """
+    return all(isinstance(cmd.op, Gate) for cmd in engine.cmd_queue)
+
+
+def is_channel(engine):
+    """Returns true if every command in the queue is either of type :class:`strawberryfields.ops.Gate`
+    or type :class:`strawberryfields.ops.Channel`.
+
+    Args:
+        engine (Engine): a Strawberry Fields engine instance
+
+    Returns:
+        bool: returns True if all queued operations are either a quantum gate or a channel
+    """
+    return all(isinstance(cmd.op, (Channel, Gate)) for cmd in engine.cmd_queue)
+
+
+def _vectorize_dm(tensor):
+    """Given a tensor with 4N indices of dimension :math:`D` each, it returns the vectorized
+    tensor with 4 indices of dimension :math:`D^N` each. This is the inverse of the procedure
+    given by :func:`_unvectorize_dm`.
+    Caution: this private method is intended to be used only for Choi and Liouville operators.
+
+    For example, :math:`N=2`,
+    ::
+        0 --|‾‾‾‾|-- 1
+        2 --|    |-- 3
+        4 --|    |-- 5
+        6 --|____|-- 7
+
+    goes to
+    ::
+        (0,2) --|‾‾‾‾|-- (1,3)
+        (4,6) --|____|-- (5,7)
+
+    Args:
+        tensor (array): a tensor with :math:`4N` indices of dimension :math:`D` each
```

Comment:
That's not the best choice of name on my part. I just wanted to indicate that it's not acting on vectors, but it's not acting on density matrices either. Feel free to change the name (or just remove "_dm" since the name is already beginning with an underscore)

Reply:
Ok, I removed the "_dm" termination